tests/test_scraper.py

```diff
@@ -130,3 +135,320 @@ def test_save_image_file(self):
+
+
+class GetLinkListTestCase(TestCase):
+
+    def test_get_link_list__no_links(self):
+        list = get_link_list('(10,.+html)$', 
+        BeautifulSoup(
+        b'''
+        <html>
+            <body>
+                <p>This is a paragraph.</p>
+                <p>This is another paragraph.</p>
+            </body>
+        </html>
+        ''',
+        'html.parser'))
+        for link in list:
+            assert link is None
+
+
+    def test_get_link_list__find_all_links(self):
+        list = get_link_list('(10,.+html)$', 
+        BeautifulSoup(
+        b'''
+        <html>
+            <body>
+                <p>This is a paragraph.</p>
+                <p>This is another paragraph.</p>
+                <a href='10,271.html'>
+                <a href='10,161.html'>
+                <a href='10,261.html'>
+                <a href='10,291.html'>
+            </body>
+        </html>
+        ''',
+        'html.parser'))
+        for link in list:
+            assert link is not None
+
+class ChangeToSummaryLinkTestCase(TestCase):
+
+    def test_change_to_summary_link(self):
+        assert change_to_summary_link(
+            "https://www.wiz.pl/10,269.html") == 'https://www.wiz.pl/19,269.html'
+
+
+class GetEditionDateTestCase(TestCase):
+
+    def test_get_edition_date__no_edition_date(self):
+        assert get_edition_date(BeautifulSoup(
+        b'''
+        <html>
+            <body>
+                <p>This is a paragraph.</p>
+                <p>This is another paragraph.</p>
+                <a href='10,271.html'>
+                <a href='10,161.html'>
+                <a href='10,261.html'>
+                <a href='10,291.html'>
+            </body>
+        </html>
+        ''',
+        'html.parser')) is None
+
+
+    def test_get_edition_date__is_edition_date(self):
+        assert get_edition_date(BeautifulSoup(
+        '''
+        <html>
+            <body>
+                <p>This is a paragraph.</p>
+                <p>This is another paragraph.</p>
+                <a href='10,271.html'>
+                <a href='10,161.html'>
+                <a href='10,261.html'>
+                <a href='10,291.html'>
+                <div style="margin-top: 0px;">
+                    <div>Wiedza i Życie  03/2019</div>
+                </div>
+            </body>
+        </html>
+        ''',
+        'html.parser')) == '03/2019'
+
+
+class GetTableOfContentsTestCase(TestCase):
+
+    def test_get_table_of_contents(self):
+        table = BeautifulSoup(
+        '''
+            <div style="margin-bottom: 15px">
+                <strong>Biometeorologia<br></strong>Siódme poty; Andrzej Hołdys<br><br>
+                <strong>Fizjologia</strong><br>Kriorewolucja; Katarzyna Kornicka<br><br>
+                <strong>Technika</strong><br>Węsząca elektronika; Justyna Jońca<br><br>
+            </div>
+        ''',  # noqa
+        'html.parser')
+
+        assert get_table_of_contents(table)[0] == dict({
```

Comment:
`dict` to jest zbędne to bo to już jest `dict`. `dict(a=3) == {'a': 3}` daje `True`